hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/portable/PortableInternalGenericRecord.java

```diff
@@ -685,13 +803,67 @@ public Object readObjectFromArray(String fieldName, int index) {
         }
     }
 
+
+    private <T> T readObjectFromArrayField(@Nonnull String fieldName, FieldType fieldType,
+                                           Reader<ObjectDataInput, T> reader, int index) {
+        int currentPos = in.position();
+        try {
+            int position = readPosition(fieldName, fieldType);
+            if (isNullOrEmpty(position)) {
+                return null;
+            }
+            in.position(position);
+            int len = in.readInt();
+            if (len == Bits.NULL_ARRAY_LENGTH || len == 0 || len <= index) {
+                return null;
+            }
+
+            int offset = in.position();
+            int pos = in.readInt(offset + index * Bits.INT_SIZE_IN_BYTES);
+            if (pos != -1) {
```

Comment:
Is this check also unnecessary?

Reply:
Removed this check as well. leftover as before.